crates/pera-core/src/consensus_handler.rs

```diff
@@ -355,6 +356,63 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
+                    if let ConsensusTransactionKind::SignatureMPCOutput(
+                        value,
+                        session_id,
+                        sender_address,
+                    ) = &transaction.kind
+                    {
+                        info!(
+                            "Received proof mpc output from authority {:?} for session {:?}",
+                            authority_index, session_id
+                        );
+
+                        let mut signature_mpc_manager = self.epoch_store.proof_mpc_manager.get();
+                        let is_valid_transaction = match signature_mpc_manager {
+                            Some(mpc_manager) => {
+                                let signature_mpc_manager = mpc_manager.lock().await;
+                                match signature_mpc_manager.try_verify_output(value, session_id) {
+                                    Ok(is_valid) => is_valid,
+                                    Err(e) => {
+                                        error!(
+                                            "Error verifying ProofMPCOutput output from session {:?} and party {:?}: {:?}",
+                                            session_id, authority_index, e
+                                        );
+                                        false
+                                    }
+                                }
+                            }
+                            None => {
+                                // TODO (#250): Make sure that the MPC manager is initialized before MPC events emitted.
+                                error!("MPC manager was not initialized when verifying ProofMPCOutput output from session {:?}", session_id);
+                                false
+                            }
+                        };
+
+                        if is_valid_transaction {
+                            let transaction =
+                                VerifiedTransaction::new_signature_mpc_output_system_transaction(
+                                    SignatureMPCOutput {
+                                        session_id: *session_id,
+                                        sender_address: *sender_address,
+                                        value: value.clone(),
+                                    },
+                                );
+                            let transaction = VerifiedExecutableTransaction::new_system(
+                                transaction,
+                                self.epoch(),
+                            );
+                            transactions.push((
+                                empty_bytes.as_slice(),
+                                SequencedConsensusTransactionKind::System(transaction),
+                                consensus_output.leader_author_index(),
+                            ));
+                        }
```

Comment:
@ycscaly IIUC, the above code handles the case where `signature_mpc_manager.try_verify_output` errors, not when it returns `Ok(false)`. In the off chance I am misunderstanding and the above code does handle the `Ok(false)` case, we should consider removing this `if is_valid_transaction` clause, since it serves no purpose?

Reply:
I agree with Erik, here there is no `else` block to catch the `Ok(false)` that happens when a malicious Validator sends
a correctly formed `ProofMPCOutput` message that contains an invalid MPC Output.

Reply:
Added todo + issue